src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs

```diff
@@ -85,9 +88,21 @@ public void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEve
 
             try
             {
+                this.discoveryCriteria = discoveryCriteria;
                 EqtTrace.Info("TestDiscoveryManager.DoDiscovery: Background test discovery started.");
                 this.testDiscoveryEventsHandler = eventHandler;
 
+                // For netcore/fullclr both packages and sources are same thing, 
+                // For UWP the actual source(exe) differs from input source(.appxrecipe) which we call package.
+                // So in such models we check if they differ, then we pass this info to test host to update TestCase source with package info,
+                // since this is needed by IDE's to map a TestCase to project.
+
+                // For older testhost(netcore), we were not passing this Source Info from runner, so adding a null check.
+                if(this.discoveryCriteria.Sources != null)
+                {
+                    this.updateTestCaseSource = this.discoveryCriteria.Sources.Except(this.discoveryCriteria.AdapterSourceMap.FirstOrDefault().Value).Any();
```

Comment:
FirstOrDefault can return null?

Reply:
The AdapterMap contains the map of "_none_" to actual sources, this cannot be null ever, & currently always contain one entry.
Though now changing logic to get test sources via querying map of "_none_"